File: Divide-Array-Into-Equal-Pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int num : nums) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }
        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            if (entry.getValue() % 2 == 1) {
                return false;
            }
        }
        return true;
    }
}

File: Divide-Array-Into-Equal-Pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        unordered_map<int,int> freq;
        for(int i:nums){
            ++freq[i];
        }
        for(pair<int,int> p:freq){
            if(p.second & 1 == 1)return false;
        }
        return true;
    }
};